### PR TITLE
Gradient Boost Regression

### PR DESCRIPTION
**Pull request by TechieHustle**
Implemented regression and added Iris dataset test.
Used Parijat's `DecisionTreeRegressor` class fit. Had to make the `predict()` function public to use it in Gradient boost regression.

### CODE REVIEW

File: algorithms/linfa-ensemble/Cargo.toml [hunk 1]
```diff
@@ -30,7 +30,10 @@ linfa-datasets = { version = "0.7.0", path = "../../datasets/", features = ["iri
 ndarray = { version = "0.15" , features = ["rayon", "approx"]}
 ndarray-rand = "0.14"
 rand = { version = "0.8", features = ["small_rng"] }
+approx = {version = "0.5"}
```
Comment (Sin317) [inline algorithms/linfa-ensemble/Cargo.toml:33]: do we need this module?

File: algorithms/linfa-ensemble/src/gradient_boost/algorithm.rs [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+            num_trees,
+            learning_rate,
+            max_features,
+            max_depth,
+            min_samples_split,
+            init_train_target_mean: 0.0,
+        }
+    }
+
+    pub fn fit(&mut self, features: &Array2<f64>, targets: &Array1<f64>) {
```
Comment (Sin317) [inline algorithms/linfa-ensemble/src/gradient_boost/algorithm.rs:40]: please add source for the algorithm in a comment

File: algorithms/linfa-ensemble/src/gradient_boost/algorithm.rs [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+            .mean()
+            .unwrap()
+            .sqrt();
+
+        println!("RMSE: {:?}", rmse);
+    }
+}
+
+
+// fn softmax(residuals: &mut Vec<f64>) -> Result<()> {
```
Comment (Sin317) [inline algorithms/linfa-ensemble/src/gradient_boost/algorithm.rs]: remove the comments if not needed

File: algorithms/linfa-ensemble/src/xgboost/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# XGBoost Tree Regression Based on [Video](https://www.youtube.com/watch?v=OtD8wVaFm6E&ab_channel=StatQuestwithJoshStarmer)
```
Comment (Sin317) [inline algorithms/linfa-ensemble/src/xgboost/README.md:1]: Remove this file since we're not supporting xgboost

File: algorithms/linfa-ensemble/Cargo.toml (hunk 1 above)
Comment (TechieHustle) [reply algorithms/linfa-ensemble/Cargo.toml:33]: Yes, in case of regressor we need to have a tolerance to approximate the values to compare if they are equal.

**Review by pkawale — APPROVED** [review]
After feedback from Khushi, I think we are good to go.

File: algorithms/linfa-ensemble/src/gradient_boost/algorithm.rs (hunk 3 above)
Comment (TechieHustle) [reply algorithms/linfa-ensemble/src/gradient_boost/algorithm.rs]: Done, it was just the code which I had written in the beginning.

File: algorithms/linfa-ensemble/src/gradient_boost/algorithm.rs (hunk 2 above)
Comment (TechieHustle) [reply algorithms/linfa-ensemble/src/gradient_boost/algorithm.rs:40]: Done

**Review by Sin317 — APPROVED** [review]
LGTM
